vcsclient/fs.go

```diff
@@ -200,6 +197,28 @@ func GetFileWithOptions(fs vfs.FileSystem, path string, opt GetFileOptions) (*Fi
 	return &fwr, nil
 }
 
+// readDir uses the passed vfs.FileSystem to read from starting at the base path. If
+// shouldRecurse is set to true, it will recurse into all sub-folders and return the
+// full sub-tree.
+func readDir(fs vfs.FileSystem, base string, shouldRecurse bool) ([]*TreeEntry, error) {
```

Comment:
Are there edge cases that I'm not thinking of hat need to be handled? If not, it's fine either way I think. Either way your call really, just wanted to mention it because it's a useful lib

Sent from my iPhone

> On May 14, 2015, at 22:30, Gabriel Aszalos <notifications@github.com> wrote:
> 
> In vcsclient/fs.go:
> 
> > @@ -200,6 +197,28 @@ func GetFileWithOptions(fs vfs.FileSystem, path string, opt GetFileOptions) (*Fi
> >  	return &fwr, nil
> >  }
> >  
> > +// readDir uses the passed vfs.FileSystem to read from starting at the base path. If
> > +// shouldRecurse is set to true, it will recurse into all sub-folders and return the
> > +// full sub-tree.
> > +func readDir(fs vfs.FileSystem, base string, shouldRecurse bool) ([]*TreeEntry, error) {
> When you left the comment I assumed this was already a dependency that we are using in the vcsstore - do you reckon it's worth adding in?
> 
> —
> Reply to this email directly or view it on GitHub.
>